src/fe-small-advertisements/src/components/App.tsx

```diff
@@ -0,0 +1,32 @@
+import React, { ChangeEvent, FormEvent, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { changeNameActionCreator } from '../actions';
+import { RootState } from '../reducers';
+
+function App() {
+  const { name } = useSelector((state: RootState) => state.user);
+  const dispatch = useDispatch();
+  const [newName, setNewName] = useState('');
+
+  const handleNewName = (e: ChangeEvent<HTMLInputElement>): void => {
+    setNewName(e.target.value);
+  };
```

Comment:
I dont think we need this function here. Can you try use `setNewName` directly on the `onChange`?

Reply:
I agree with you. I also think it is not necessary to create an extra function to control and input like these one. I did it because Typescript makes put a lot of extra code and it looks very weird to me. But I have to get used to it. I put it directly in the onChange method, it is not that bad.

Reply:
you dont _always_ have to annotate. Is most useful to annotate modules that are reusable and have interfaces so that when we refactor something we have clean interfaces. In this case not annotating wont change that much imo.